ultraplot/figure.py

```diff
@@ -1131,11 +1126,83 @@ def _add_subplot(self, *args, **kwargs):
         kwargs.setdefault("label", f"subplot_{self._subplot_counter}")
         kwargs.setdefault("number", 1 + max(self._subplot_dict, default=0))
         kwargs.pop("refwidth", None)  # TODO: remove this
+
         ax = super().add_subplot(ss, _subplot_spec=ss, **kwargs)
+        self._WARN_SHARING_GEOAXIS = True
+        # Allow sharing for GeoAxes if rectilinear
+        if self._sharex or self._sharey:
+            if (
+                len(self.axes) > 1
+                and isinstance(ax, paxes.GeoAxes)
+                and not ax._is_rectilinear()
+            ):
+                self._unshare_axes()
+                if self._WARN_SHARING_GEOAXIS:
```

Comment:
We should be using the [warnings filter `once`](https://docs.python.org/3/library/warnings.html#the-warnings-filter) here instead of rolling our own global. 

The code as written warns once per object, not once per class as well.

Reply:
Ah thanks was not aware of this.

Reply:
Wouldn't this filter all warnings once?

Reply:
This does not happen. Removing the variable